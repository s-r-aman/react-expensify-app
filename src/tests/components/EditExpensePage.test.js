import React from 'react';
import { shallow } from "enzyme";

import { EditExpensePage } from "../../components/EditExpensePage";
import expenses from "../fixtures/expenses";

let wrapper,editExpense, removeExpense, history;

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = {push: jest.fn()};
  wrapper = shallow(
    <EditExpensePage 
      editExpense={editExpense} 
      removeExpense={removeExpense} 
      history={history} 
      expense={expenses[0]}
    />
  );
});

describe('<EditExpensePage/>', () => {

  test("should render the Edit Expense Component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should handle the on submit prop properly", () => {
    wrapper.find("ExpenseForm").prop("onSubmit")(expenses[1]);
    expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[1]);
  });

  test('should handle the on remove prop properly', () => {
    wrapper.find("button").prop("onClick")();
    expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[0].id})    
  });
});