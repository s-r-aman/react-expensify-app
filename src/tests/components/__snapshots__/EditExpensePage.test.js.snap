// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditExpensePage/> should render the Edit Expense Component correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
